jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/exceptions/OperationNotSupportedException.java

```diff
@@ -0,0 +1,8 @@
+package uk.gov.homeoffice.digital.sas.jparest.exceptions;
+
+public class OperationNotSupportedException extends IllegalArgumentException {
```

Comment:
Looks like this maps to returning a 400 Bad Request.
Just wanna double check that's the most suitable for an unsupported patch operation.
@stephenkaye-ho do you have any thoughts, 405 possible candidate?

Reply:
It's not a bad shout and understand why a 405 was suggested but it is referring explictly to the HTTP methods and a server returning a 405  must generate an Allow header field in a 405 status code response. (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405)
The Allow header contains a comma-separated list of allowed HTTP request methods e.g. GET, PUT, POST 
(https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow)
With that in mind I feel a 400 fits better

Reply:
yep that makes sense, 400 sounds good 👍